napari/layers/_source.py

```diff
@@ -22,12 +25,15 @@ class Source(BaseModel):
         `viewer.open_sample`.
     widget: FunctionGui, optional
         magicgui widget, if the layer was added via a magicgui widget.
+    parent: Layer, optional
+        parent layer if the layer is a duplicate.
     """
 
     path: Optional[str] = None
     reader_plugin: Optional[str] = None
     sample: Optional[Tuple[str, str]] = None
     widget: Optional[FunctionGui] = None
+    parent: Optional[Layer] = None
```

Comment:
it should be `weakref` to not keep the layer in memory if deleted from interface.

Reply:
It is now handled in the `parent` validator